### PR TITLE
Workflow improvements

### PR DESCRIPTION
**Pull request by j0sh**
* Set devenv specific protocol parameters. Fixes https://github.com/livepeer/devenv/issues/29
* Reuses existing transcoder/broadcaster/geth data if a node is down.
* Expose geth to the world
* Remove gasLimit usage
* Initialize rounds automatically
* Use shared libs for some deps

cc @darkdarkdragon

### CODE REVIEW

**Comment by j0sh** [comment]
Exposed geth to the world. Makes it easier to set up geth in the VM then do local development by connecting the local node's `ethUrl` to the VM.

File: dot_lpdev_cmds.sh [hunk 1]
```diff
@@ -245,8 +245,11 @@ function __lpdev_geth_run {
 
   echo "Running Geth miner with the following command:
   geth -networkid 54321 \\
-       -rpc \\
+       -rpc -ws \\
+       -rpcaddr "0.0.0.0" -wsaddr "0.0.0.0" \\
+       -rpccorsdomain "*" -wsorigins "*" \\
        -rpcapi 'personal,account,eth,web3,net' \\
+       -wsapi 'personal,account,eth,web3,net' \\
```
Comment (darkdarkdragon) [inline dot_lpdev_cmds.sh]: you haven't added these parameters to the command that actually runs `geth`

Comment (j0sh) [reply dot_lpdev_cmds.sh]: Ah yep, fixed

**Comment by j0sh** [comment]
Reuse the geth mining account where possible instead of creating a new one on each geth startup.

Link against shared libraries to reduce build times. Static linking should still be used for releases.

**Pull request merged**